lib/HTML/FormBuilder/Base.pm

```diff
@@ -73,9 +73,10 @@ sub _build_element_and_attributes {
 
         # skip attributes that are not intended for HTML
         next if ($key =~ /^(?:option|text|hide_required_text|localize|wrap_in_div_class)/i);
-        if ($attributes->{$key}) {
-            $html .= ' ' . $key . '="' . $attributes->{$key} . '"';
-        }
+        next if not defined $attributes->{$key};
+        next if '' . $attributes->{$key} eq '';
```

Comment:
Do we really need the concat to `''`?  
```suggestion
        next if $attributes->{$key} eq '';
```

Since the previous `next if` already excludes the `undef` case, a string here may only be either a scalar value (that can upgrade to string) or ''.